### PR TITLE
Issue/v4/983 implement documents news endpoints

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR solves #983 

 * CRUD actions on `/documents`, `/news` and other object type endpoints are implemented

 * `/home` response updated with new dynamic endpoints

### CODE REVIEW

File: plugins/BEdita/API/src/Controller/ObjectsController.php [hunk 1]
```diff
@@ -61,12 +90,94 @@ public function index()
...
+     *
+     * @return void
+     * @throws \Cake\Network\Exception\BadRequestException Throws an exception if submitted data is invalid.
+     */
+    public function add()
+    {
+        $this->request->allowMethod('post');
+
+        $object = $this->Objects->newEntity($this->request->data);
+        $object->type = $this->request->data('type');
```
Comment (batopa) [inline plugins/BEdita/API/src/Controller/ObjectsController.php:119]: I think that here is required a check on consistency between `data('type')` and the endpoint used. For example I could send
```http
POST /documents

{
    "type": "users",
    "attributes": {
        "title": "New document"
    }
}
```

A `400 Bad Request` should be expected (missing unit test too)

File: plugins/BEdita/API/src/Controller/ObjectsController.php [hunk 2]
```diff
@@ -61,12 +90,94 @@ public function index()
...
+     * @throws \Cake\Network\Exception\ConflictException Throws an exception if object ID in the payload doesn't match
+     *      the URL object ID.
+     * @throws \Cake\Network\Exception\NotFoundException Throws an exception if specified object could not be found.
+     * @throws \Cake\Network\Exception\BadRequestException Throws an exception if submitted data is invalid.
+     */
+    public function edit($id)
+    {
+        $this->request->allowMethod('patch');
+
+        if ($this->request->data('id') != $id) {
```
Comment (batopa) [inline plugins/BEdita/API/src/Controller/ObjectsController.php:151]: Same validation data on `type` shown above is required

File: plugins/BEdita/API/src/Controller/ObjectsController.php [hunk 3]
```diff
@@ -61,12 +90,94 @@ public function index()
...
+     *
+     * @param int $id object ID.
+     * @return void
+     * @throws \Cake\Network\Exception\InternalErrorException Throws an exception if an error occurs during deletion.
+     */
+    public function delete($id)
+    {
+        $this->request->allowMethod('delete');
+
+        $object = $this->Objects->get($id, [
```
Comment (batopa) [inline plugins/BEdita/API/src/Controller/ObjectsController.php:186]: Missing conditions on object type if specific endpoint is used and 404 Not Found if missing the object to delete

**Review by batopa — CHANGES_REQUESTED** [review]

File: plugins/BEdita/API/src/Controller/ObjectsController.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/src/Controller/ObjectsController.php:119]: Right, type check added

File: plugins/BEdita/API/src/Controller/ObjectsController.php (hunk 2 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/src/Controller/ObjectsController.php:151]: Type check added in edit

File: plugins/BEdita/API/src/Controller/ObjectsController.php (hunk 3 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/src/Controller/ObjectsController.php:186]: Type check added here too.
404 not found is handled directly in `$this->Objects->get()` call
Added 404 tests for delete and edit actions

**Review by batopa — APPROVED** [review]

**Pull request merged**